### DISCUSSION TITLE
Embed image into an exisitng PDF file

### DISCUSSION THREAD

**Discussion by nk-alex** [QuestPDF/QuestPDF#908] [Q&A] [answered]
I have seen #521, this question is more or less related. 

Having an already exisitng PDF file, I would like to embed an image in the specified pages. Is this achievable using QuestPDF?

**Comment by girlpunk** [comment]
Unfortunately QuestPDF doesn't offer any functionality for editing or otherwise opening existing PDFs.

**Comment by DawidRobertNowak** [comment, marked as answer]
Hello,

We are closing this thread due to inactivity. Thank you for taking the time to reach out and contribute to the project — your feedback helps us improve QuestPDF.

While QuestPDF remains primarily a PDF generation library, the `DocumentOperation` API (introduced in version 2024.12.0) now supports overlaying and underlaying content onto existing PDF pages, which may help achieve your use case. Full documentation is available here:
https://www.questpdf.com/concepts/document-operations.html

Should you encounter any further problems, please don't hesitate to open a new issue.

Thank you!
